library/src/scripts/components/SelectBox.tsx

```diff
@@ -0,0 +1,140 @@
+/*
+ * @author Stéphane LaFlèche <stephane.l@vanillaforums.com>
+ * @copyright 2009-2018 Vanilla Forums Inc.
+ * @license GPL-2.0-only
+ */
+
+import * as React from "react";
+import classNames from "classnames";
+
+import { getRequiredID } from "../componentIDs";
+import { t } from "../application";
+import DropDownItemButton from "@library/components/dropdown/items/DropDownItemButton";
+import { checkCompact, downTriangle } from "@library/components/Icons";
+import DropDown from "@library/components/dropdown/DropDown";
+import { ButtonBaseClass } from "@library/components/forms/Button";
+
+export interface ISelectBoxItem {
+    name: string;
+    className?: string;
+    onClick: () => {};
+    selected?: boolean;
+    outdated?: boolean;
+}
+
+interface IProps {
+    className?: string;
+    id?: string;
+    children: ISelectBoxItem[];
+    stickTop?: boolean; // Adjusts the flyout position vertically
+    stickRight?: boolean; // Adjusts the flyout position horizontally
+}
+
+export interface ISelfLabelledProps extends IProps {
+    label: string;
+}
+
+export interface IExternalLabelledProps extends IProps {
+    describedBy: string;
+}
+
+interface IState {
+    id: string;
+    selectedIndex: number;
+    selectedItem: any;
+}
+
+/**
+ * Generates Select Box component (similar to a select)
+ */
+export default class SelectBox extends React.Component<ISelfLabelledProps | IExternalLabelledProps, IState> {
+    public static defaultProps = {
+        selectedIndex: 0,
+    };
+
+    public constructor(props) {
+        super(props);
+
+        this.state = {
+            id: getRequiredID(props, "selectBox-"),
+            selectedIndex: props.selectedIndex,
+            selectedItem: props.selectedItem || props.children[props.selectedIndex],
+        };
+    }
+
+    /**
+     * Handle click on item in select box.
+     * @param selectedItem data for item
+     * @param index the index of the item
+     */
+    public handleClick = (selectedItem: ISelectBoxItem, index: number) => {
+        this.setState({
+            selectedIndex: index,
+            selectedItem,
+        });
+    };
+
+    public render() {
+        const space = `&nbsp;`;
+
+        const selectItems =
+            this.props.children && this.props.children.length > 0
```

Comment:
This if check here is largely unnecessary. Mapping over an empty array will return an empty array, which react will render the same way that it renders `null` or `undefined`.